src/vs/platform/terminal/common/terminal.ts

```diff
@@ -391,7 +393,7 @@ export interface IShellLaunchConfig {
 	 * The codicon ID to use for this terminal. If not specified it will use the default fallback
 	 * icon.
 	 */
-	icon?: string;
+	iconPath?: ThemeIcon | Codicon | string | URI | { light: URI, dark: URI };
```

Comment:
Let's stick with `icon` everywhere on the main side, keep `iconPath` just for the extension API.